Added environmental variables

Chris-green-stfc opened the pull request:
Added different environments to make deploying easier

Chris-green-stfc committed:
Added environmental variables

Added different environments to make deploying easier

jose-caballero reviewed (changes requested):
Could you replace with the new block?

{% set total_n_users = number_of_users | int %}
{% set width_n_users = ("%d" % total_n_users).len() %}
{% for i in range (1, (number_of_users | int) + 1) %}
- {{ username }}-{{ "%0*d" % (width_n_users, i) }}
{% endfor %}

DavidFair commented on roles/deploy_jhub/templates/config.yaml.j2:
```diff
@@ -3,16 +3,16 @@ proxy:
     letsencrypt:
-      # Uncomment acmeserver for a test certificate
-      # acmeServer: "https://acme-staging-v02.api.letsencrypt.org/directory"
+{% if acme_server is true %}
```
Nit: This should be `staging_cert` as you're basically saying

`acme_server == True  # Gets a staging cert`

DavidFair replied:
We should the enter an else with the non-staging cert, since running this with `acme_server` true then running it with it set false won't reset the value, it will leave it set to: `"https://acme-staging-v02.api.letsencrypt.org/directory"`

DavidFair reviewed (changes requested):
Partial review

DavidFair commented on vars/secrets.yaml:
I don't think we need this? AFAIK we don't have any secrets since `secretToken` is generated

Chris-green-stfc committed:
Added inventory

Added inventory
added group vars
removed secrets.yaml
fixed acme server bug when redeploying without staging cert

meoflynn reviewed (dismissed):
LGTM 

I tested it on a dev cluster and it works
